tests/lib/rules/prefer-const.js

```diff
@@ -56,6 +56,12 @@ ruleTester.run("prefer-const", rule, {
         "let a; function foo() { if (a) {} a = bar(); }",
         "let a; function foo() { a = a || bar(); baz(a); }",
         "let a; function foo() { bar(++a); }",
+        "let predicate; [typeNode.returnType, predicate] = foo();",
+        "let predicate; let rest; [typeNode.returnType, predicate, ...rest] = foo();",
```

Comment:
Shouldn't these be invalid since they have the default `destructuring: "any"` option?